tests/test_shell_launcher.py

```diff
@@ -0,0 +1,323 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import pathlib
+import shutil
+import subprocess
+import unittest.mock
+
+import psutil
+import pytest
+
+from smartsim._core.shell.shellLauncher import ShellLauncher, ShellLauncherCommand, sp
+from smartsim._core.utils import helpers
+from smartsim._core.utils.shell import *
+from smartsim.entity import _mock, entity
+from smartsim.error.errors import LauncherJobNotFound
+from smartsim.status import JobStatus
+
+# TODO tests bad vars in Popen call at beginning
+# tests -> helper.exe : pass in None, empty str, path with a space at beginning, a non valid command
+#       -> write a test for the invalid num of items - test_shell_launcher_fails_on_any_invalid_len_input
+#       -> have border tests for 0,1,4,6 cmd vals -> work correctly without them -> raise ValueError
+# do all of the failures as well as the sucess criteria
```

Comment:
Is this TODO still needed? I think I saw _some_ of this being tested, but not all. Is this something we want to address in this PR or a follow on. If a follow on, is there a ticket somewhere for this work?

Reply:
1) invalid num of items is not required given that it was written down during the era of unpacking a tuple and expecting the length to be #.
2) I do not believe border tests for cmd vals is necessary given the new NamedTuple
3) can probably add a check for popen cwd